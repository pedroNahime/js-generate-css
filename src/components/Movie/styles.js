import styled from 'styled-components'

export const Content = styled.a`
  display: flex;
  flex-direction: column;
  margin-left: 20px;
  width: 260px;
  text-decoration: none;
  margin-bottom: 45px;

  img {
    height: 250px;
  }

  strong {
    font-size: 13px;
    margin-top: 10px;
    color: #585858;
  }

  p {
    line-height: 22px;
    margin-top: 5px;
    font-size: 13px;
    color: #393939;
  }

  &:hover img {
    opacity: 0.4;
  }
`
