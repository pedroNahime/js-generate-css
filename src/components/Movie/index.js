import PropTypes from 'prop-types'
import React from 'react'
import { Content } from './styles'

const Movie = props => {
  const renderMovies = () => {
    if (props.movie.media_type === 'movie') {
      return (
        <>
          <img
            src={`https://image.tmdb.org/t/p/original/${props.movie.poster_path}`}
            alt={props.movie.title}
          />
          <strong>{props.movie.title}</strong>
        </>
      )
    } else if (props.movie.media_type === 'tv') {
      return (
        <>
          <img
            src={`https://image.tmdb.org/t/p/original/${props.movie.poster_path}`}
            alt={props.movie.title}
          />
          <strong>{props.movie.name}</strong>
        </>
      )
    }
  }
  return <Content>{renderMovies()}</Content>
}

export default Movie

Movie.propTypes = {
  movie: PropTypes.object.isRequired
}
