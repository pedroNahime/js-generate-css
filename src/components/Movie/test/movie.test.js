import React from 'react'
import { render } from '@testing-library/react'
import Movie from '../index'

const mockProps = {
  movie: {
    media_type: 'movie',
    poster_path: 'some_path',
    title: 'some_title'
  }
}

describe('<Header/>', () => {
  test('render the component', () => {
    const { container } = render(<Movie {...mockProps} />)
    expect(container).toMatchSnapshot()
  })
})
