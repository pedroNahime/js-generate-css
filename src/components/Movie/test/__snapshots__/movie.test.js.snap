// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> render the component 1`] = `
<div>
  <a
    class="sc-bdVaJa jKuiS"
  >
    <img
      alt="some_title"
      src="https://image.tmdb.org/t/p/original/some_path"
    />
    <strong>
      some_title
    </strong>
  </a>
</div>
`;
